### ISSUE TITLE
ERROR: null

### ISSUE THREAD

**Issue by IvanVodkevich** [mehboss/CustomRecipes#3]
Hello there, MehBoss!
I'm sincerely sorry for bothering you again with another bug, but the 1.7.4 version does not seem to work. When pasting it into the server's plugin folder through an SFTP protocol, a folder called "CustomRecipes" appeared, indicating that the plugin should work.
However, as you may guess, it doesn't. When loading in, the only message that the server sends is:

_[21:50:22 INFO]: [CustomRecipes] Enabling CustomRecipes v1.7.4
[21:50:22 INFO]: [CustomRecipes] Made by MehBoss on Spigot. For support please PM me and I will get back to you as soon as possible!
[21:50:22 ERROR]: Error occurred while enabling CustomRecipes v1.7.4 (Is it up to date?)
[21:50:22 INFO]: [CustomRecipes] Disabling CustomRecipes v1.7.4_

But, the confusion doesn't end there!
Not only is the _blacklisted.yml_ file empty, but the commands such as _/crecipe_ and _/crecipes_ don't seem to properly work, either. They're there, the game's console expects them, but such commands as _/crecipe reload_ don't work.
When initiating these commands, such reports show up in the console:

_[21:51:08 INFO]: IvanVodkevich issued server command: /crecipe reload
[21:51:08 ERROR]: null
[21:57:09 INFO]: IvanVodkevich issued server command: /crecipes IvanVodkevich
[21:57:09 ERROR]: null_

The config.yml file contains the recipes and nothing seems to be wrong with it. However, the rest is doodoo.
Is this an issue with my server and its host? I'm running a 24/7 PaperSpigot 1.19.4

Thank you for your time, and I apologize again.

**Comment by mehboss** [comment]
Hello again. I am sincerely sorry for the issues you are encountering. Do you happen to have a FULL console error log that includes any line numbers I can refer to -- since I am having trouble replicating this? I am using my test server on server.pro with a fresh install of paper 1.19.4 and it starts up and runs CustomRecipes perfectly. Again, I am sorry for any inconvenience.

**Comment by mehboss** [comment]
This is my startup log for paper 1.19.4. As you can see, I can not replicate your issue, unfortunately. Are you in offline mode by any chance? And is that the only error in console and the ENTIRE error?
`[INFO] Loaded 7 recipes

[INFO] Starting minecraft server version 1.19.4

[INFO] Loading properties

[INFO] This server is running Paper version git-Paper-545 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: bacbf86)

[INFO] Server Ping Player Sample Count: 12

[INFO] Using 4 threads for Netty based IO

[WARNING] [!] The timings profiler has been enabled but has been scheduled for removal from Paper in the future.

We recommend installing the spark profiler as a replacement: https://spark.lucko.me/

For more information please visit: https://github.com/PaperMC/Paper/issues/8948

[INFO] [ChunkTaskScheduler] Chunk system is using 1 I/O threads, 1 worker threads, and gen parallelism of 1 threads

[INFO] Default game type: SURVIVAL

[INFO] Generating keypair

[INFO] .... Starting Minecraft server on *:41766

[INFO] .... Using epoll channel type

[INFO] .... Paper: Using libdeflate (Linux x86_64) compression from Velocity.

[INFO] .... Paper: Using OpenSSL 3.0.x (Linux x86_64) cipher from Velocity.

[INFO] .... [CustomRecipes] Loading server plugin CustomRecipes v1.7.4

[INFO] .... Server permissions file permissions.yml is empty, ignoring it

[INFO] .... Preparing level "world"

[INFO] .... Preparing start region for dimension minecraft:overworld

[INFO] .... Time elapsed: 237 ms

[INFO] .... Preparing start region for dimension minecraft:the_nether

[INFO] .... Time elapsed: 107 ms

[INFO] .... Preparing start region for dimension minecraft:the_end

[INFO] .... Time elapsed: 111 ms

[INFO] .... [CustomRecipes] Enabling CustomRecipes v1.7.4

[INFO] .... [CustomRecipes] Made by MehBoss on Spigot. For support please PM me and I will get back to you as soon as possible!

[INFO] .... Starting GS4 status listener

[INFO] .... Thread Query Listener started

[INFO] .... Query running on 0.0.0.0:41766

[INFO] .... Running delayed init tasks

[INFO] .... Done (4.321s)! For help, type "help"

[INFO] .... Timings Reset

[INFO] .... [CustomRecipes] Checking for updates..

[WARNING] .... [CustomRecipes] An update has been found! This could be bug fixes or additional features. Please update CustomRecipes at https://www.spigotmc.org/resources/authors/mehboss.139036/

[INFO] ....>.... Checking version, please wait...

[INFO] ....>.... Unknown command. Type "/help" for help.

[INFO] .... This server is running Paper version git-Paper-545 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: bacbf86)

You are running the latest version

[INFO] ....>.... Server Plugins (1):

[INFO] .... Bukkit Plugins:

[INFO] .... - CustomRecipes`

**Comment by IvanVodkevich** [comment]
Hello-hello. I apologize for writing so late, but I've just logged in to check on this thread for your response.

As you said, I've initiated the _/version_ command and here are the only results:
_[15:52:11 INFO]: Checking version, please wait...
[15:52:11 INFO]: This server is running Paper version git-Paper-527 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 6884092)
You are 18 version(s) behind
Download the new version at: https://papermc.io/downloads/paper
Previous version: git-Paper-491 (MC: 1.19.4)_

Answering one of your questions: Yes, I am on offline mode, so that my non-premium friends can join and play on the server.

Here're full console logs that pop up when I restart the server:
_[15:56:21 INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[15:56:23 INFO]: Loaded 7 recipes
[15:56:24 INFO]: Starting minecraft server version 1.19.4
[15:56:24 INFO]: Loading properties
[15:56:24 INFO]: This server is running Paper version git-Paper-527 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 6884092)
[15:56:24 INFO]: Server Ping Player Sample Count: 12
[15:56:24 INFO]: Using 4 threads for Netty based IO
[15:56:25 WARN]: [!] The timings profiler has been enabled but has been scheduled for removal from Paper in the future.
We recommend installing the spark profiler as a replacement: https://spark.lucko.me/
For more information please visit: https://github.com/PaperMC/Paper/issues/8948
[15:56:25 INFO]: [ChunkTaskScheduler] Chunk system is using 1 I/O threads, 2 worker threads, and gen parallelism of 2 threads
[15:56:25 INFO]: Default game type: SURVIVAL
[15:56:25 INFO]: Generating keypair
[15:56:25 INFO]: Starting Minecraft server on 0.0.0.0:25688
[15:56:25 INFO]: Using epoll channel type
[15:56:25 INFO]: Paper: Using libdeflate (Linux x86_64) compression from Velocity.
[15:56:25 INFO]: Paper: Using OpenSSL 1.1.x (Linux x86_64) cipher from Velocity.
[15:56:26 INFO]: [SkinsRestorer] Loading server plugin SkinsRestorer v14.2.10
[15:56:26 INFO]: [Timber] Loading server plugin Timber v1.0-SNAPSHOT
[15:56:26 INFO]: [CatchBall] Loading server plugin CatchBall v1.1.7
[15:56:26 INFO]: [CustomRecipes] Loading server plugin CustomRecipes v1.7.4
[15:56:26 INFO]: Server permissions file permissions.yml is empty, ignoring it
---
---
---
---
[15:56:26 INFO]: Preparing level "duckworld"
[15:56:26 INFO]: Preparing start region for dimension minecraft:overworld
[15:56:27 INFO]: Time elapsed: 361 ms
[15:56:27 INFO]: Preparing start region for dimension minecraft:the_nether
[15:56:27 INFO]: Time elapsed: 165 ms
[15:56:27 INFO]: Preparing start region for dimension minecraft:the_end
[15:56:27 INFO]: Time elapsed: 89 ms
[15:56:27 INFO]: [SkinsRestorer] Enabling SkinsRestorer v14.2.10
[15:56:29 INFO]: [SkinsRestorer] Detected Minecraft v1_19_R3, using MappingSpigotSkinRefresher.
[15:56:30 INFO]: [SkinsRestorer] [ACF] Enabled Asynchronous Tab Completion Support!
[15:56:30 INFO]: [SkinsRestorer] Using paper join listener!
[15:56:30 INFO]: [Timber] Enabling Timber v1.0-SNAPSHOT
[15:56:30 INFO]: [CatchBall] Enabling CatchBall v1.1.7
[15:56:32 INFO]: [CatchBall] Plugin is already the latest version
[15:56:32 INFO]: [CustomRecipes] Enabling CustomRecipes v1.7.4
[15:56:32 INFO]: [CustomRecipes] Made by MehBoss on Spigot. For support please PM me and I will get back to you as soon as possible!
[15:56:32 ERROR]: Error occurred while enabling CustomRecipes v1.7.4 (Is it up to date?)
[15:56:32 INFO]: [CustomRecipes] Disabling CustomRecipes v1.7.4
[15:56:34 INFO]: Loaded 7 recipes
[15:56:36 INFO]: Starting GS4 status listener
[15:56:36 INFO]: Thread Query Listener started
[15:56:36 INFO]: Running delayed init tasks
[15:56:36 INFO]: Query running on 0.0.0.0:25688
[15:56:36 INFO]: Done (11.407s)! For help, type "help"
[15:56:36 INFO]: Timings Reset
[15:56:36 INFO]: [SkinsRestorer] ----------------------------------------------
[15:56:36 INFO]: [SkinsRestorer] +==================+
[15:56:36 INFO]: [SkinsRestorer] | SkinsRestorer |
[15:56:36 INFO]: [SkinsRestorer] |------------------|
[15:56:36 INFO]: [SkinsRestorer] | Standalone Mode |
[15:56:36 INFO]: [SkinsRestorer] Current version: 14.2.10
[15:56:36 INFO]: [SkinsRestorer] This is the latest version!_

I've deleted four lines of the logs that I didn't want to show in public. They don't contain anything related to CustomRecipes or any other plugins.

**Comment by mehboss** [comment]
Hello again. I have just tried in offline-mode, and it works. I am still unable to duplicate this error. I would advise you to update your paper.jar as you are 18 versions behind and see if that resolves your issue. If it does not, please let me know.

**Comment by IvanVodkevich** [comment]
Good afternoon.
To make it quick, I downloaded the _latest_ version of Paper and it's still not working. Here're the newest logs:

_[12:24:27 INFO]: [CustomRecipes] Enabling CustomRecipes v1.7.4
[12:24:27 INFO]: [CustomRecipes] Made by MehBoss on Spigot. For support please PM me and I will get back to you as soon as possible!
[12:24:27 ERROR]: Error occurred while enabling CustomRecipes v1.7.4 (Is it up to date?)
[12:24:27 INFO]: [CustomRecipes] Disabling CustomRecipes v1.7.4_

_[12:24:56 INFO]: Checking version, please wait...
[12:24:57 INFO]: This server is running Paper version git-Paper-527 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 6884092)_

I'm not sure about the Java, but the server was freshly updated about 3 days ago and everything except Paper was updated. So I _think_ it should be the newest version? I don't know how to check it, admittedly.

Maybe reach out to me on Discord? That way we can communicate more efficiently.
IvanVodkevich#2327

Thanks in advance.

**Comment by mehboss** [comment]
Hello, I have added you on discord. Also, I would advise to upload your paperspigot version, as you are still 18 versions behind.

[15:52:11 INFO]: This server is running Paper version git-Paper-527 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 6884092)
You are 18 version(s) behind

https://papermc.io/downloads/paper
Try #545 or newer. You are on #527.

**Comment by IvanVodkevich** [comment]
Hello, once again.

I come with only one issue this time: You've sent me the Discord friend request, but being the dumb donkey I am, I've declined your request instead of accepting it as I was in a rush. I waited in hopes of you re-sending your friend request again, but nothing happened so far, so I ask you to send me a friend request. I won't decline this time, but in case I will, keep sending it, lol

Thank you for your time and cooperation, I'm constantly looking forward to fix this plugin issue!
